cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerConfigService.java

```diff
@@ -203,4 +209,94 @@ private String getRoleConfigGroupNameByTypeAndServiceName(String roleType, Strin
+            return servicesResourceApi.readServices(clusterName, DataView.SUMMARY.name());
+        } catch (ApiException e) {
+            LOGGER.error("Failed to get %s service name from Cloudera Manager.", e);
+            return new ApiServiceList();
+        }
+    }
+
+    public void stopClouderaManagerService(ApiClient client, StackDtoDelegate stack, String serviceType) throws CloudbreakException {
+        ServicesResourceApi servicesResourceApi = clouderaManagerApiFactory.getServicesResourceApi(client);
+        LOGGER.info("Trying to stop services for service type: {} for cluster {}", serviceType, stack.getName());
+        getServiceName(stack.getName(), serviceType, servicesResourceApi)
+                .ifPresentOrElse(
+                        stopServices(stack, servicesResourceApi, client),
+                        () -> {
+                            LOGGER.info("{} service name is missing, skiping stop.", serviceType);
+                            throw new ClouderaManagerOperationFailedException(String.format("Service of type: %s is not found", serviceType));
+                        });
+    }
+
+    private Consumer<String> stopServices(StackDtoDelegate stack, ServicesResourceApi servicesResourceApi, ApiClient client) {
+        ApiServiceConfig apiServiceConfig = new ApiServiceConfig();
+        return serviceName -> {
+            try {
+                servicesResourceApi.stopCommand(stack.getName(), serviceName);
+            } catch (Exception e) {
+                LOGGER.error("Failed to stop services for service type: {} for cluster {}", serviceName, stack.getName(), e);
+                throw new ClouderaManagerOperationFailedException(e.getMessage(), e);
+            }
+        };
+    }
+
+    public void startClouderaManagerService(ApiClient client, StackDtoDelegate stack, String serviceType) throws CloudbreakException {
+        ServicesResourceApi servicesResourceApi = clouderaManagerApiFactory.getServicesResourceApi(client);
+        LOGGER.info("Trying to start services for service type: {} for cluster {}", serviceType, stack.getName());
+        getServiceName(stack.getName(), serviceType, servicesResourceApi)
+                .ifPresentOrElse(
+                        startServices(stack, servicesResourceApi, client),
+                        () -> {
+                            LOGGER.info("{} service name is missing, skiping start.", serviceType);
+                            throw new ClouderaManagerOperationFailedException(String.format("Service of type: %s is not found", serviceType));
+                        });
+    }
+
+    private Consumer<String> startServices(StackDtoDelegate stack, ServicesResourceApi servicesResourceApi, ApiClient client) {
+        ApiServiceConfig apiServiceConfig = new ApiServiceConfig();
+        return serviceName -> {
+            try {
+                servicesResourceApi.startCommand(stack.getName(), serviceName);
```

Comment:
could you please add a line of log here for that we're gonna execute a command?

Reply:
done